scripts/evaluator.py

```diff
@@ -0,0 +1,460 @@
+        if self.config.device is None:
+            raise ValueError("No device provided")
+        # Use constant for default checkpoint directory.
+        self.checkpoint_path: str = (
+            self.config.checkpoint_path or CHECKPOINT_DEFAULT_DIR
+        )
+        os.makedirs(os.path.dirname(self.checkpoint_path), exist_ok=True)
+
+        self.netuid = self.config.netuid
+        self.subtensor = bt.subtensor(config=self.config)
+        self.metagraph = self.subtensor.metagraph(netuid=self.netuid)
+        self.hparams = tplr.load_hparams()
+        self.wallet = bt.wallet(config=self.config)
+
+        # Mock for the comms class
+        self.uid = 1
+
+        self.model = LlamaForCausalLM(config=self.hparams.model_config)
+        self.model.to(self.config.device)
+
+        self.tokenizer = self.hparams.tokenizer
+        self.comms = tplr.comms.Comms(
+            wallet=self.wallet,
+            save_location="/tmp",
+            key_prefix="model",
+            config=self.config,
+            netuid=self.netuid,
+            metagraph=self.metagraph,
+            hparams=self.hparams,
+            uid=self.uid,
+        )
+
+        self.buckets = self.comms.get_all_buckets()
+        self.last_eval_window = 0
+        self.stop_event = asyncio.Event()
+        self.last_block_number = 0
+
+        # Initialize metrics logger with consistent patterns
+        self.metrics_logger = tplr.metrics.MetricsLogger(
+            prefix="E",
+            uid=str(self.uid),
+            config=self.config,
+            role="evaluator",
+            group="evaluations",
+            job_type="eval",
+        )
+
+    async def update_state(self) -> None:
+        """
+        Refresh the metagraph and bucket information.
+        """
+        self.metagraph = self.subtensor.metagraph(netuid=self.netuid)
+        self.buckets = self.comms.get_all_buckets()
+
+    async def load_latest_model(self) -> Tuple[bool, dict, int, int]:
+        """Load and prepare the latest model checkpoint for evaluation.
+
+        This method:
+        1. Fetches the latest checkpoint from storage
+        2. Verifies checkpoint validity
+        3. Loads model weights and momentum
+        4. Updates internal state trackers
+
+        Returns:
+            Tuple containing:
+            - success (bool): Whether loading succeeded
+            - checkpoint_data (dict): Checkpoint metadata
+            - checkpoint_window (int): Window number of checkpoint
+            - global_step (int): Global training step
+        """
+        result = await self.comms.get_latest_checkpoint()
+        tplr.logger.info(f"[DEBUG] get_latest_checkpoint() result: {result}")
+        if not result:
+            tplr.logger.error(
+                f"No valid checkpoints found. Check bucket: {getattr(self.comms, 'bucket_name', 'unknown')}, "
+                f"key_prefix: {self.comms.key_prefix}"
+            )
+            return (False, {}, 0, 0)
```

Comment:
Should we not first check if there is a new checkpoint, and if it is we download it?